### ISSUE TITLE
NO_COLOR/MB_COLORIZE_LOGS does not remove all ansi codes

### ISSUE THREAD

**Issue by jimmy-outschool** [metabase/metabase#13538]
In an attempt to remove all non-plain text bits from the logs (emojis and ansi codes) the following environment variables were set as shown.

    MB_COLORIZE_LOGS=false
    MB_EMOJI_IN_LOGS=false

This removes all the _color_, but does not remove all the ansi codes which presumably is the point. Instead the following is produced:

    10-20 15:29:45 [1mINFO metabase.util[0m :: Loading Metabase...

The colors used after the `::` are not present, but the bold around the log level and location remain. From the docker image these are controlled via `/app/metabase.jar/log4j.properties`, specifically the following snippet:

```properties
# log to the console
log4j.appender.console=org.apache.log4j.ConsoleAppender
log4j.appender.console.Target=System.out
log4j.appender.console.layout=org.apache.log4j.PatternLayout
log4j.appender.console.layout.ConversionPattern=%d{MM-dd HH:mm:ss} \u001b[1m%p %c{2}\u001b[0m :: %m%n
```

Changing the pattern to the following resolves the problem:

```properties
log4j.appender.console.layout.ConversionPattern=%d{MM-dd HH:mm:ss} %p %c{2} :: %m%n
```

Instead it seems reasonable to expect a non-development image to either have ansi output disabled or at least an option to do so. Some logging tools do not work properly when ansi codes are present and most tools do not support displaying the colors which are instead presented as muck in the text.

For my case, this blocks extracting metrics to be used to automatically restart to work around metabase bugs. I am currently working around this with a [modified image containing a tweaked `log4j.properties` config in the `metabase.jar`](https://github.com/jimmy-outschool/metabase-plain-logs).

It is also not clear where the magic is happening in the docker build the build blocks do not seem to be present in this repository.

**Comment by camsaul** [comment]
> It is also not clear where the magic is happening in the docker build the build blocks do not seem to be present in this repository.

For Metabase 0.37.0+ we've upgraded to Log4j 2. So we have a `log4j2.xml` file now instead of `log4j.properties`. That's why you're not seeing it in the `master` branch. Take a look at the `release-0.36.x` branch to see the `log4j.properties` file we shipped prior to 0.37.0

**Comment by camsaul** [comment]
As you've noticed, `NO_COLOR` or `MB_COLORIZE_LOGS` only affects the parts of the log output Metabase generates directly, and not the stuff that Log4j does based on its config. I think making the log4j uncolorized by default makes sense. Mind opening a PR?
